packages/text-bitmap/src/BitmapText.ts

```diff
@@ -301,53 +331,199 @@ export class BitmapText extends Container
         }
 
         const lenChars = chars.length;
-        const tint = this.tint;
+
+        const pagesMeshData: Record<number, PageMeshData> = {};
+
+        // new Map<BaseTexture, PageMeshData>();
+        const meshesToAdd: Mesh[] = [];
 
         for (let i = 0; i < lenChars; i++)
         {
-            let c = this._glyphs[i]; // get the next glyph sprite
+            const texture = chars[i].texture;
+            const baseTextureUid = texture.baseTexture.uid;
 
-            if (c)
+            if (!pagesMeshData[baseTextureUid])
             {
-                c.texture = chars[i].texture;
+                let pageMeshData = pageMeshDataPool.pop();
+
+                if (!pageMeshData)
+                {
+                    const geometry = new MeshGeometry();
+                    const material = new MeshMaterial(Texture.EMPTY);
+
+                    const mesh = new Mesh(geometry, material);
+
+                    pageMeshData = {
+                        index: 0,
+                        indexCount: 0,
+                        vertexCount: 0,
+                        uvsCount: 0,
+                        total: 0,
+                        mesh,
+                        vertices: null,
+                        uvs: null,
+                        indices: null,
+                    };
+                }
+
+                // reset data..
+                pageMeshData.index = 0;
+                pageMeshData.indexCount = 0;
+                pageMeshData.vertexCount = 0;
+                pageMeshData.uvsCount = 0;
+                pageMeshData.total = 0;
+                // TODO need to get page texture here somehow..
+                pageMeshData.mesh.texture = new Texture(texture.baseTexture);
+                pageMeshData.mesh.tint = this._tint;
+
+                meshesToAdd.push(pageMeshData.mesh);
+
+                pagesMeshData[baseTextureUid] = pageMeshData;
             }
-            else
+
+            pagesMeshData[baseTextureUid].total++;
+        }
+
+        const currentMeshes = this._currentMeshes;
+
+        for (let i = 0; i < currentMeshes.length; i++)
+        {
+            if (meshesToAdd.indexOf(currentMeshes[i]) === -1)
```

Comment:
`pagesMeshData` was never saved (i.e. was initialized with `{}`), so I'm having trouble seeing how `meshesToAdd` and `currentMeshes` having something in common.

Reply:
If this is correct, maybe some inline comments on how it works could be beneficial.

Reply:
Good question! the thing we need to keep track of is the mesh's as they need to be added removed (potentially!) on a text change.

For that the `meshesToAdd` is where all the meshes we are that must be added to the stage are added. We then compare that with `_currentMeshes` and remove ones that shouldn't be there, whilst adding ones that should.

Then after we have done that, `meshesToAdd` becomes the new `_currentMeshes` list.

Will add some comments to make a bit clearer 👍